File: Programação Para Dispositivos Móveis/Exercícios em Sala App/Aula 08/CalcularIMC/app/src/main/java/com/pedro/calcularimc/MainActivity.java
package com.pedro.calcularimc;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        EditText textoPeso = findViewById(R.id.editTextPeso);
        EditText textoAltura = findViewById(R.id.editTextAltura);
        Button botaoCalcular = findViewById(R.id.buttonCalcular);
        TextView textResultado = findViewById(R.id.editTextResultado);

        botaoCalcular.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String campoPeso = textoPeso.getText().toString();
                String campoAltura = textoAltura.getText().toString();

                if (!campoPeso.isEmpty() && !campoAltura.isEmpty()) {
                    double peso = Double.parseDouble(campoPeso);
                    double altura = Double.parseDouble(campoAltura);
                    double imc = peso / (altura*altura);

                    if(imc < 18.5) textResultado.setText(String.format("Seu imc é: Peso Baixo"));
                    else if(imc >= 18.5 && imc <= 24.9) textResultado.setText(String.format("Seu imc é: Peso Normal"));
                    else if(imc >= 25.0 && imc <= 29.9) textResultado.setText(String.format("Seu imc é: Sobrepeso"));
                    else if(imc >= 30.0 && imc <= 34.9) textResultado.setText(String.format("Seu imc é: Obesidade I"));
                    else if(imc >= 35.0 && imc <= 39.9) textResultado.setText(String.format("Seu imc é: Obesidade II"));
                    else textResultado.setText(String.format("Seu imc é: Obesidade III"));

                } else {
                    Toast.makeText(getApplicationContext(), "Preencha todos os campos", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: Programação Para Dispositivos Móveis/Simulado App/Simulado Gabriel/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        val letraMin = "abcdefghijklmnopqrstuvwxyz"
        val numero = "0123456789"
        val caracter = "!@#$%&"

        val botaoResultado: Button = findViewById(R.id.button)
        val gerarSenha: TextView = findViewById(R.id.gerarSenha)

        botaoResultado.setOnClickListener {
            val senha = StringBuilder()

            for (i in 0 until 3) {
                val letra = letraMin.random()
                val num = numero.random()
                val simbolo = caracter.random()
                senha.append(letra)
                senha.append(num)
                senha.append(simbolo)
            }

            gerarSenha.text = senha.toString()
        }
    }
}
